client/landing/stepper/declarative-flow/internals/steps-repository/sensei-plan/index.tsx

```diff
@@ -0,0 +1,396 @@
+
+	const planObject = supportedPlans.find( ( plan ) => {
+		return plan && 'business' === plan.periodAgnosticSlug;
+	} );
+
+	const handlePlanBillingPeriodChange = ( newBillingPeriod: Plans.PlanBillingPeriod ) => {
+		setBillingPeriod( newBillingPeriod );
+	};
+
+	const woothemesProduct = useSelect(
+		( select ) => {
+			const yearly = select( PRODUCTS_LIST_STORE ).getProductBySlug( 'woothemes_sensei_yearly' );
+			const monthly = select( PRODUCTS_LIST_STORE ).getProductBySlug( 'woothemes_sensei_monthly' );
+
+			if ( ! yearly || ! monthly ) {
+				return {
+					monthlyPrice: 0,
+					yearlyPrice: 0,
+					price: 0,
+					productSlug: '',
+					currencyCode: '',
+				};
+			}
+
+			return {
+				monthlyPrice: monthly.cost,
+				yearlyPrice: yearly.cost,
+				price: billingPeriodIsAnnually ? Math.ceil( yearly.cost / 12 ) : monthly.cost,
+				productSlug: billingPeriodIsAnnually ? yearly.product_slug : monthly.product_slug,
+				currencyCode: yearly.currency_code,
+			};
+		},
+		[ billingPeriodIsAnnually ]
+	);
+
+	const planProduct = useSelect(
+		( select ) => {
+			const monthly = select( PLANS_STORE ).getPlanProduct(
+				planObject?.periodAgnosticSlug,
+				'MONTHLY'
+			);
+			const yearly = select( PLANS_STORE ).getPlanProduct(
+				planObject?.periodAgnosticSlug,
+				'ANNUALLY'
+			);
+			if ( ! yearly || ! monthly ) {
+				return {
+					monthlyPrice: 0,
+					yearlyPrice: 0,
+					price: 0,
+					productSlug: '',
+				};
+			}
+
+			return {
+				monthlyPrice: monthly.rawPrice,
+				yearlyPrice: yearly.rawPrice,
+				price: billingPeriodIsAnnually ? Math.ceil( yearly.rawPrice / 12 ) : monthly.rawPrice,
+				productSlug: '',
+			};
+		},
+		[ billingPeriodIsAnnually, planObject ]
+	);
+
+	const goToDomainStep = useCallback( () => {
+		if ( goToStep ) {
+			goToStep( 'senseiDomain' );
+		}
+	}, [ goToStep ] );
+
+	const onPlanSelect = async () => {
+		try {
+			setStatus( Status.Bundling );
+
+			// Wait for a bit to get an animation in the beginning of site creation.
+			await new Promise( ( res ) => setTimeout( res, 100 ) );
+
+			setProgress( {
+				percentage: 25,
+				title: siteProgressTitle,
+			} );
+
+			await createSenseiSite( {
+				username: currentUser?.username || '',
+				languageSlug: locale,
+				visibility,
+			} );
+
+			setProgress( {
+				percentage: 50,
+				title: cartProgressTitle,
+			} );
+
+			const newSite = getNewSite();
+			const siteId = newSite?.blogid;
+			setSelectedSite( newSite?.blogid );
+			await setIntentOnSite( newSite?.site_slug as string, SENSEI_FLOW );
+			await saveSiteSettings( newSite?.blogid as number, { launchpad_screen: 'full' } );
+
+			setProgress( {
+				percentage: 75,
+				title: styleProgressTitle,
+			} );
+
+			if ( siteId ) {
+				const selectedStyleVariationTitle = getSelectedStyleVariation()?.title;
+				const styleVariations: StyleVariation[] = await getStyleVariations( siteId, 'pub/course' );
+				const theme: Theme = await getSiteTheme( siteId, 'pub/course' );
```

Comment:
I'd parallelize this as well.

Reply:
Done b9dce195869c24f0eed172a9f71e162af98a778a